Fixed various offsets in TS

reaperrr opened the pull request:
Successor to #11334.

No changes, I just messed up #11334 while cleaning my repo.

reaperrr committed:
Fix Component Tower weapon and muzzle offsets

reaperrr committed:
Fix TS Temple of Nod art offset

reaperrr committed:
Fix TS stealth tank firing offsets

reaperrr committed:
Set firing offset and enable muzzle flash on TS Titan

pchote commented on mods/ts/rules/gdi-support.yaml:
```diff
@@ -101,27 +101,27 @@ GACTWR:
 		UpgradeMinEnabledLevel: 1
 		Weapon: RedEye2
-		LocalOffset: 512,0,512
+		LocalOffset: 192,0,1152
```
This looks wrong ingame to me - it comes from the middle of the turret instead of the front.

pchote commented on mods/ts/rules/gdi-vehicles.yaml:
```diff
@@ -158,8 +158,11 @@ MMCH:
+		MuzzleSequence: muzzle
 		Recoil: 127
 		RecoilRecovery: 26
+		LocalOffset: 512, 224, 784
```
This also looks quite far offset:
![screen shot 2016-05-28 at 18 33 38](https://cloud.githubusercontent.com/assets/167819/15628802/cf21ff5a-2502-11e6-81ea-4bbe9e77e5c0.png)

pchote replied:
You should check whether the facing fudge has been disabled.

reaperrr committed:
Fix Nod SAM Site sequence and armament offsets

The turret sequence in particular was located too far to the left.

reaperrr committed:
Fixed InitialFacings of TS defenses to match original

reaperrr commented:
Updated:
- fixed Titan offset, `ClassicFacingFudge` was indeed not disabled
- fixed GDI SAM turret firing offset

Additionally,
- fixed Nod SAM Site offsets (sequences and armament)
- fixed `InitialFacing` of all defense to face north-east (was considered "north" in original TS)

reaperrr commented:
I just realised the InitialFacings need to be 224, not 192. Will fix that tomorrow/later today.

reaperrr commented:
Updated.

reaperrr commented:
@Mailaender Since I added the InitialFacing fixes after your review, I guess you should check that commit to validate the :+1:

pchote commented:
Looks fine to me now.  If you enable the combat overlay the firing positions for the sam site aren't right, but that cancels out the effect of the otherwise badly positioned smoke trail so i'm not going to block the pr over that.

reaperrr commented:
>  If you enable the combat overlay the firing positions for the sam site aren't right, but that cancels out the effect of the otherwise badly positioned smoke trail so i'm not going to block the pr over that.

Yeah, that's intentional.

Mailaender commented:
Thanks. Looking good. :white_check_mark:

Phrohdoh commented:
An incorrect overlay is intentional?

reaperrr commented:
@Phrohdoh No, the larger-than-necessary distance of the firing offsets from the turret is intentional, because it looks better in practice.
